### PR TITLE
fix: gitignore :wrench:

### PR DESCRIPTION
**Pull request by robertoschiavone**
# Description

A fresh gitignore. Despite having way more lines, it's easier to maintain since it's autogenerated, and it covers more tools, editors, operating systems.

I, for example, use `pyenv` and I have to constantly remind myself not to accidentally commit my local `.python-version`. Now I don't have to worry anymore 🙃 

# Checklist:

- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by RedTachyon** [comment]
We probably don't need a fully automatically-generatable .gitignore, so please make sure that all the entries that were present previously, are also present now. Some of them seem rather specific to gymnasium, like vizdoom.ini, rllab_mujoco etc. so they won't get automatically picked up by the toptal generator. While I don't know the exact story behind the inclusion of each of those items, we should assume they're here for a reason and not remove them without a good reason otherwise.

File: .gitignore [hunk 1]
```diff
@@ -1,43 +1,416 @@
...
+Thumbs.db
+Thumbs.db:encryptable
+ehthumbs.db
+ehthumbs_vista.db
+
+# Dump file
+*.stackdump
 
-junk
-MUJOCO_LOG.txt
```
Comment (Kallinteris-Andreas) [inline .gitignore:31]: should not be removed

File: .gitignore [hunk 2]
```diff
@@ -1,43 +1,416 @@
...
+# Folder config file
+[Dd]esktop.ini
 
-rllab_mujoco
```
Comment (Kallinteris-Andreas) [inline .gitignore:33]: should not be removed

**Review by Kallinteris-Andreas — COMMENTED** [review]
I am generally against 'generic' `.gitignore` files, only project specific `temp/cache` files should be excluded, it is the responsibility of the developer to edit it locally to ignore your IDE files

**Pull request closed without merging**